reader_test.go

```diff
@@ -1047,6 +1053,54 @@ func testReaderConsumerGroupReadContentAcrossPartitions(t *testing.T, ctx contex
 	}
 }
 
+// Build a struct to implement the ReadPartitions interface.
+type MockConnWatcher struct {
+	count int
+	partitions [][]Partition
+}
+func (m *MockConnWatcher) ReadPartitions(topics ...string) (partitions []Partition, err error) {
+	partitions = m.partitions[m.count]
+	m.count++
```

Comment:
Unfortunately, this is the type of project where there are lots of time-based unit tests.  Our approach is to make sure that things eventually end up in a consistent state, and usually to deal with CI noisiness, add a lot of extra time.  One thing we could do here to make the test more resilient is to cap `m.count` at `len(m.partitions)-1`.  That way, the test will be idempotent even if the watcher comes in a couple more times.

Reply:
That makes sense! Never thought about that. 
Changes pushed